cmd/node/app_test.go

```diff
@@ -305,45 +304,64 @@ func (suite *AppTestSuite) validateBlocksAndATXs(untilLayer types.LayerID) {
 	}
 
 	// assuming all nodes have the same results
-	layers_per_epoch := int(suite.apps[0].Config.LayersPerEpoch)
-	layer_avg_size := suite.apps[0].Config.LayerAvgSize
+	layerAvgSize := suite.apps[0].Config.LayerAvgSize
 	patient := datamap[suite.apps[0].nodeId.Key]
 
-	lastlayer := len(patient.layertoblocks)
+	lastLayer := len(patient.layertoblocks)
 
-	total_blocks := 0
+	totalBlocks := 0
 	for _, l := range patient.layertoblocks {
-		total_blocks += len(l)
+		totalBlocks += len(l)
 	}
 
-	first_epoch_blocks := 0
-
-	for i := 0; i < layers_per_epoch; i++ {
+	firstEpochBlocks := 0
+	for i := 0; i < layersPerEpoch; i++ {
 		if l, ok := patient.layertoblocks[types.LayerID(i)]; ok {
-			first_epoch_blocks += len(l)
+			firstEpochBlocks += len(l)
 		}
 	}
 
-	total_atxs := 0
-
-	for _, atxs := range patient.atxPerEpoch {
-		total_atxs += int(atxs)
+	// assert number of blocks
+	totalEpochs := int(untilLayer.GetEpoch(uint16(layersPerEpoch))) + 1
+	lateNodeActiveEpochs := totalEpochs - int(startInLayer.GetEpoch(uint16(layersPerEpoch))) - 2
+	allMiners := len(suite.apps)
+	//exp := (layerAvgSize/(allMiners-1)*(allMiners-1)*(totalEpochs-lateNodeActiveEpochs) + layerAvgSize/(allMiners-1)*allMiners + (layerAvgSize/allMiners)*allMiners*(lateNodeActiveEpochs-2)) * layersPerEpoch
+	exp := (layerAvgSize*layersPerEpoch)/(allMiners-1)*(allMiners-1)*(totalEpochs-lateNodeActiveEpochs-1) + (layerAvgSize*layersPerEpoch)/(allMiners-1)*allMiners
```

Comment:
Please use the commented formula, with revisions to the first and last summands: add -1 after 'lateNodeActiveEpochs'. Insert 'layersPerEpoch' to each summand.
If we change the test parameters, this is the formula we should use!

Reply:
Fixed